### ISSUE TITLE
Easier modification of arrays/maps

### ISSUE THREAD

**Issue by Keats** [Keats/tera2#40]
Right now it has to be done via filters but we can probably figure out a way to support it better in the language directly

**Comment by asimpletune** [comment]
> One thing is that I don't know how common updating a single field in a map is, I think most users don't update maps/object at all?

I don't think so either. The `set foo.bar` syntax proposed above (and its array counterpart) to me looks good but is also made redundant via the spread syntax, which covers those cases and much more. However I think it is something that should be kept in mind when tera2 adoption continues in the future. Could be a useful ergonomic feature to add one day. 

It's hard to know for sure since the ability to even create maps in Zola in the first place wasn't even possible until `load_data` - with literal option - was only added a few years back, and I think tera is mostly used via Zola. 

Still, spread is very generic and flexible, so it makes sense to keep the feature set limited if you already get a lot of expressivity this way.
